Chore/lint fix

osadi opened the pull request:
@ollema Most of these changes should be safe and only touch types. But: there are a few `if`-statements that I'm uncertain about. Types say they are unnecessary, but I guess they were there for a reason 🙂

~Edit: wow that failed spectacularly~

ollema commented:
looks great!

there is one caveat though - I don't know if you have worked with shadcn(-svelte) before but it's not a traditional component library where you install a package and import components.

instead you use a cli to install and update component code that you own.

this has it's benefits and drawbacks. see https://next.shadcn-svelte.com/docs for more information.

in this case, it means that most of the code in `src/lib/components/ui` are actually managed "upstream" so to speak. when you update components you actually get a diff that you should read through and make sure that you understand what changes are made.

I have overriden some of the components and added my own. but in general it's much easier to work this shadcn if you keep modifications minimal. therefore I don't think it's great to make changes only for the sake of the linter, since that add unnecessary overhead.

if there are linter issues in "vendored code" then I think it's better to add an exception in eslint to ignore said rule for the `src/lib/components/ui` folder. we can trust that this code is well written anyways IMHO.

HOWEVER, when I run:
```
pnpm dlx shadcn-svelte@next update --all --yes && pnpm format
```
which is how I usually update the components and format them I get a giant diff.

this is likely due to the tailwind 4 migration. I can tackle that another day.

--

sorry for the wall of text but what I suggest is that you rollback any changes in `$lib/src/components/ui` and instead modify the eslint config to ignore this.

osadi commented:
No problem. I'll revert all in `ui/*`. I thought that shadcn was copy/paste and owned at the moment it was pasted in to the project. You live and you learn 🙂

ollema commented:
> No problem. I'll revert all in `ui/*`. I thought that shadcn was copy/paste and owned at the moment it was pasted in to the project. You live and you learn 🙂

that is true, we do own the code.

but I think it's nice being able to also easily adapt upstream bug fixes if you can without that much work.

so if you can avoid it I think it's nice to be able to avoid making modifications. but when needed you can, as I have done with e.g. `src/lib/components/ui/form/index.ts` and `src/lib/components/ui/sonner/sonner.svelte`.

both of these actually change the functionality of the component (or adds more custom components in the barrel file). but in my opinion it's harder to motivate changes that causes diffs when you don't modify the functionality and instead only conform to linting rules.

does that make sense or am I crazy?

osadi commented:
Makes sense. Those changes that came with the update is not something that I would like to maintain, so can one get that for free that's perfect 🙂

osadi commented on src/lib/components/ui/data-table/render-helpers.ts:
```diff
@@ -77,8 +77,8 @@ export class RenderSnippetConfig<TProps> {
 	T extends Component<any>,
-	Props extends ComponentProps<T>
->(component: T, props: Props = {} as Props) {
+	TProps extends ComponentProps<T>
```
My bad. Missed that in the rebase.

ollema commented:
a decent acceptance criteria would be to run:
```
pnpm dlx shadcn-svelte@next update --all --yes && pnpm format
```
and see that we only get a diff in:
- `package.json` (also ignore and reset for now, it downgrades for some reason)
- `pnpm-lock.yaml` (also safe to ignore and reset)
- `src/lib/components/ui/form/index.ts` (expected since we want to change this file, so should revert)
- `src/lib/components/ui/sonner/sonner.svelte` (expected since we want to change this file, so should revert)

osadi commented:
Almost
```bash
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   package.json
        modified:   pnpm-lock.yaml
        modified:   src/lib/components/ui/form/index.ts
        modified:   src/lib/components/ui/sonner/sonner.svelte
        modified:   src/lib/utils.ts
```

`src/lib/utils.ts` should also be excluded then I guess.

ollema commented:
yeah right, that file is also something that is added by the CLI. but oddly enough not placed in `../ui/`. I guess they want to encourage usage of the `cn` util for your own components as well

ollema reviewed (approved):
very nice! now we can enable blocking linting in CI

the tanstack preset seems to include some good rules as well, great job
